app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -32,7 +32,34 @@
                     $("#all-unread-notifications-indicator").removeClass('ajax-loading'); 
                 },
                 success: function(result) {
-                    renderGlobalUnreadNotificationsIndicator(result.count)
+                    if (result.notifications && result.notifications.length > 0) {
+                        // Get the stored notifications from sessionStorage (those already notified)
+                        let storedNotifications = JSON.parse(sessionStorage.getItem('PWA_NOTIFICATIONS') || '[]');
+
+                        // Filter out notifications that have already been read
+                        const unreadNotificationIds = result.notifications.map(n => n.notificationId);
+                        const filteredStoredNotifications = storedNotifications.filter(storedNotification =>
+                            unreadNotificationIds.includes(storedNotification.notificationId)
+                        );
+
+                        // Update sessionStorage to only store unread notifications
+                        sessionStorage.setItem('PWA_NOTIFICATIONS', JSON.stringify(filteredStoredNotifications));
+
+                        // Filter out notifications that have already been notified
+                        var newNotifications = result.notifications.filter(function (notification) {
+                            return !storedNotifications.some(function (storedNotification) {
+                                return storedNotification.notificationId === notification.notificationId;
+                            });
+                        });
+
+                        // If we have new notifications, call refreshPWA with the new notifications list
+                        if (newNotifications.length > 0) {
+                            refreshPWA(result.count, newNotifications);
```

Comment:
De ce a fost nevoie sa apelezi aici `refreshPWA`? Am vazut ca face deja asta `renderGlobalUnreadNotificationsIndicator`. Am vazut ca ai adaugat acesteia parametru `notifications`, pe care insa nu il pasezi, de ce? As zice ca:

1. poti sa nu mai apelezi `refreshPWA` aici daca pasezi parametrul `newNotifications` lui `renderGlobalUnreadNotificationsIndicator`
2. `if (result.notifications && result.notifications.length > 0) {` de mai sus de ce a fost nevoie? 
3. Ma gansesc ca poti sa nu mai pui verificarea ` if (newNotifications.length > 0) {` caci an vazut ca ai pus-o deja in interiorul `refreshPWA`

Reply:
0. Da, apelam acolo `refreshPWA` si dupa nu mai tirmiteam parametrul `notifications` la functia de render pentru ca trimisesem deja notificarile. Ideea era sa nu trimit un array gol mai departe si sa nu apelez acel render de mai multe ori inutil.
1. Da, pare abordarea mai eficienta.
2. Intai testam daca array-ul trimis de controller e gol sau nu, apoi daca cel cu notificarile necitite este gol sau nu. Testarea de nil e inutila, cum am mentionat si mai sus.
3. Da, verific acelasi lucru in ambele locuri, voi pastra numai o singura verificare.

Reply:
Dar cum este acum tot se apeleaza acel `renderGlobal...` mereu. In plus ea are un parametru care nu este folosit

Reply:
Se apeleaza doar o data la fiecare rulare, nu? Asa era si inainte. Intradevar, cand nu exista notificari, acel parametru este inutil, dar nu am gasit alta solutie ca sa trimit notificarile spre PWA. Adica traseul e: `requestAndRenderGlobalUnreadNotificationsIndicator` -> `renderGlobalUnreadNotificationsIndicator` -> `refreshPWA` -> `showNotification`

Reply:
MOM. Varianta modificata este ok